### PR TITLE
create components and add second endpoint in server

### PR DESCRIPTION
**Pull request by arnold1105**
-Add a retrieve function in seed.js to get all data from the shop with the second endpoint '/shop' created in server. 
-Create ajax.js to get data
-Create shopItems, shopItemList, and shopInfo componenets
-add css styling

### CODE REVIEW

File: client/src/components/ajax.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import $ from 'jquery';
+
+const fetchShopItems = (callback) => {
+  $.get('http://localhost:3000/shop', (data) => {
+    callback(null, data);
+  });
```
Comment (elizabethcan) [inline client/src/components/ajax.js:6]: how are you requesting a specific shop's items without sending it a shop id? your endpoint should probably be `/shops/:shopId` at the minimum. If you want to be more in line with REST it would be `/shops/:shopId/items` which is basically saying "give me all the items for the shop with this id"

also, are you thinking about how you will get to a given shop when all you're going to be given is an itemId? (since we will be navigating the page according to an itemId not a shopId). I have the same setup for my component, so I will be querying my DB for the given itemId, then retrieving the storeId for that item, then querying the DB for all the reviews associated with that storeId.

File: client/src/components/ajax.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+};
+export default fetchShopItems;
```
Comment (elizabethcan) [inline client/src/components/ajax.js:8]: new line

File: client/src/components/shopInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import style from './style.css.js';
+
+const ShopInfo = (props) => {
+  return (
+    <div className="shop" style={style.shop}>
+      <img src="http://loremflickr.com/75/75/tech?lock=1"></img>
```
Comment (elizabethcan) [inline client/src/components/shopInfo.jsx:7]: don't forget to swap these out for images from you s3

File: client/src/components/shopInfo.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      <br/>
+      <div><b>Valhalla</b></div>
+      <font color='gray'>in Old Royal, Washington</font>
+    </div>
+  );
+};
+
+export default ShopInfo;
```
Comment (elizabethcan) [inline client/src/components/shopInfo.jsx:15]: new line

File: client/src/components/shopItem.jsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  return (
+    <div>
+      <img src={props.item.image} width="181" height="144" ></img>
+      <div style={style.text}>{props.item.description}</div>
+      <div style={style.price}>${props.item.price.toFixed(2)}</div>
+    </div>
+  );
+};
+  
+export default ShopItem;
```
Comment (elizabethcan) [inline client/src/components/shopItem.jsx:14]: new line

File: client/src/components/shopItemList.jsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import ShopItem from './shopItem.jsx';
+
+const ShopItemList = (props) => {
+  return (
+    <div className="items">
+      {props.items.map((item, index) => {
+        if (props.page * 8 <= index && props.page * 8 + 8 > index) {
```
Comment (elizabethcan) [inline client/src/components/shopItemList.jsx:8]: can you explain what this is doing?

Comment (elizabethcan) [reply client/src/components/shopItemList.jsx:8]: oh it's rendering that item if it belongs to that page, I think I understand

File: client/src/components/style.css.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
+export default {
+  container: {
+    display: 'flex',
+    position: 'absolute',
+    'justify-content': 'center',
```
Comment (elizabethcan) [inline client/src/components/style.css.js:5]: you don't have to change this, but you could also leave the quotes off the keys by camel casing i.e. 'justify-content' = justifyContent

File: client/src/components/style.css.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+  shop: {
+    display: 'flex',
+    'flex-direction': 'column',
+    'justify-content': 'center',
+    'align-items': 'center',
+    'margin-left': 'auto',
+    'margin-right': 'auto',
+  }
+};
```
Comment (elizabethcan) [inline client/src/components/style.css.js:57]: new line

File: client/src/components/summary.jsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+          <button className="button" style={style.button}> <b>&lt;</b> </button>
+          <button className="button" style={style.button}> <b>&gt;</b> </button>
+        </span>
+
+      </div>
+    );
+  }
+}
+
+export default Summary;
```
Comment (elizabethcan) [inline client/src/components/summary.jsx:52]: new line

File: client/src/index.jsx [hunk 10]
```diff
@@ -1,16 +1,5 @@
...
-      <div>
-        <h1>Shop</h1>
-      </div>)
-  }
-}
-
-
-ReactDOM.render(<App />, document.getElementById('summary'));
\ No newline at end of file
+ReactDOM.render(<Summary />, document.getElementById('summary'));
```
Comment (elizabethcan) [inline client/src/index.jsx:5]: new line

File: database/seed.js [hunk 11]
```diff
@@ -27,13 +27,13 @@ itemList.deleteMany({}, () => {
...
-  for (let itemId = 1; itemId <= 100; itemId ++) {
+  for (let itemId = 0; itemId <= 100; itemId ++) {
     let newItem = new itemList({
       productId: itemId, 
       shop: 'My Shop',
-      location: faker.address.city() + ', ' + faker.address.zipCode(),
+      location: faker.address.city() + ', ' + faker.address.state(),
       description: faker.lorem.sentence(), 
-      image: faker.image.fashion(),
+      image: faker.image.technics(),
```
Comment (elizabethcan) [inline database/seed.js:36]: you should be using images from s3 not faker images

File: database/seed.js [hunk 12]
```diff
@@ -62,3 +71,4 @@ let seed = () => {
 seed();
 
 module.exports.retrieveOne = retrieveOne;
+module.exports.retrieve = retrieve;
```
Comment (elizabethcan) [inline database/seed.js:74]: new line

File: database/seed.js [hunk 13]
```diff
@@ -46,6 +46,15 @@ let createSummary = function() {
   }
 };
 
+let retrieve = (callback) => {
```
Comment (elizabethcan) [inline database/seed.js:49]: I would suggest you be more specific with the naming of your functions. I'm not sure what data this is retrieving exactly. It looks like it's retrieving everything from `itemList` but I'm not sure what the data structure of itemList is, and looking at your server file below, it looks like you're hitting a `/shop` endpoint to call on this `retrieve` function. I think you are trying to get all items for a given shop. Perhaps a better name for this function would be something like `retrieveShopItems`.

**Review by elizabethcan — CHANGES_REQUESTED** [review]
As I mentioned in an earlier comment, I'm unsure of how you're going to be able to retrieve different store data depending on which item show page we are currently on. If I am missing something, please feel free to fill me in! Otherwise, I think you need to do some work to figure out the path of DB querying you actually need to do. Try to think about it in terms of the information you'll be receiving and the information you'll need to display. You're going to receive an `itemId` corresponding to the page that we're currently on on the browser. With that `itemId` you need to figure out what `storeId` that corresponds to. Then you need to get _all_ of the items that belong to that store and display those

File: client/src/components/shopInfo.jsx (hunk 3 above)
Comment (JonBeyond) [reply client/src/components/shopInfo.jsx:7]: Make sure you plan some time to learn how S3 works.  It'll take a couple hours.

File: client/src/components/style.css.js (hunk 7 above)
Comment (JonBeyond) [reply client/src/components/style.css.js:5]: I agree, and React will give you a warning here, you should use `justifyContent`

File: client/src/components/summary.jsx [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+  componentDidMount() { 
+    this.updateShopItems();    
+  }
+
+  updateShopItems() {
+    fetchShopItems((err, data) => {
+      if (err) {
+        console.log(err);
+        return;
```
Comment (JonBeyond) [inline client/src/components/summary.jsx:25]: recommend you change `return` to `throw err`

File: server/index.js [hunk 15]
```diff
@@ -17,6 +18,15 @@ app.get('/shop/:productId', (req, res) => {
   });
 });
 
+app.get('/shop/', (req, res) => {
```
Comment (JonBeyond) [inline server/index.js:21]: similar to what @elizabethcan said above, you need to change this endpoint to include the shopID.

File: client/src/components/ajax.js (hunk 1 above)
Comment (arnold1105) [reply client/src/components/ajax.js:6]: Thanks for pointing it out, i didn't think of that, I was just thinking we are only showing one shop at this time, so I didn't create the shopID for the items, instead i just assigned every items belongs to the same shop. But you are right that I'll need to be more in line with REST and structure it better for further development, I'll work on that.
